ddl/ddl_api.go

```diff
@@ -7564,6 +7584,120 @@ func checkIgnorePlacementDDL(ctx sessionctx.Context) bool {
+		err := SetDirectResourceGroupUnit(groupInfo.ResourceGroupSettings, opt.Tp, opt.StrValue, opt.UintValue)
+		if err != nil {
+			return err
+		}
+	}
+
+	switch {
+	case stmt.IfNotExists:
+		break
+	default:
+		_, ok := d.GetInfoSchemaWithInterceptor(ctx).ResourceGroupByName(groupName)
+		if ok {
+			return infoschema.ErrResourceGroupExists.GenWithStackByArgs(groupName)
+		}
+	}
+
+	if groupName.L == defaultResourceGroupName {
+		return errors.Trace(infoschema.ErrReservedSyntax.GenWithStackByArgs(groupName))
+	}
+
+	logutil.BgLogger().Debug("create resource group", zap.String("name", groupName.O), zap.Stringer("resource group settings", groupInfo.ResourceGroupSettings))
+
+	// TODO: valid the group information.
+	groupIDs, err := d.genGlobalIDs(1)
+	if err != nil {
+		return err
+	}
+	groupInfo.ID = groupIDs[0]
+
+	job := &model.Job{
+		SchemaName: groupName.L,
+		Type:       model.ActionCreateResourceGroup,
+		BinlogInfo: &model.HistoryInfo{},
+		Args:       []interface{}{groupInfo, false},
+	}
+	err = d.DoDDLJob(ctx, job)
+	err = d.callHookOnChanged(job, err)
+	return err
+}
+
+// DropResourceGroup implements the DDL interface.
+func (d *ddl) DropResourceGroup(ctx sessionctx.Context, stmt *ast.DropResourceGroupStmt) (err error) {
+	groupName := stmt.ResourceGroupName
+	is := d.GetInfoSchemaWithInterceptor(ctx)
+	// Check group existence.
+	group, ok := is.ResourceGroupByName(groupName)
+	if !ok {
+		err = infoschema.ErrResourceGroupNotExists.GenWithStackByArgs(groupName)
+		if stmt.IfExists {
+			ctx.GetSessionVars().StmtCtx.AppendNote(err)
+			return nil
+		}
+		return err
+	}
+
+	job := &model.Job{
+		SchemaID:   group.ID,
+		SchemaName: group.Name.L,
+		Type:       model.ActionDropResourceGroup,
+		BinlogInfo: &model.HistoryInfo{},
+		Args:       []interface{}{groupName},
+	}
+	err = d.DoDDLJob(ctx, job)
+	err = d.callHookOnChanged(job, err)
+	return err
+}
+
+func buildResourceGroup(oldGroup *model.ResourceGroupInfo, options []*ast.ResourceGroupOption) (*model.ResourceGroupInfo, error) {
+	groupInfo := &model.ResourceGroupInfo{Name: oldGroup.Name, ID: oldGroup.ID, ResourceGroupSettings: &model.ResourceGroupSettings{}}
+	for _, opt := range options {
+		err := SetDirectResourceGroupUnit(groupInfo.ResourceGroupSettings, opt.Tp, opt.StrValue, opt.UintValue)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return groupInfo, nil
+}
+
+// AlterResourceGroup implements the DDL interface.
+func (d *ddl) AlterResourceGroup(ctx sessionctx.Context, stmt *ast.AlterResourceGroupStmt) (err error) {
+	groupName := stmt.ResourceGroupName
+	is := d.GetInfoSchemaWithInterceptor(ctx)
+	// Check group existence.
+	group, ok := is.ResourceGroupByName(groupName)
+	if !ok {
+		return infoschema.ErrResourceGroupNotExists.GenWithStackByArgs(groupName)
+	}
+	newGroupInfo, err := buildResourceGroup(group, stmt.ResourceGroupOptionList)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if newGroupInfo.Name.L == defaultResourceGroupName {
```

Comment:
How about allow the user to config the quota of the default resource group, we can still provide a default settings if the user doesn't config it. BTW, `DropResourceGroup` does not contains this check.

Reply:
> What about let the parser only accept either RU mode or Naive mode but not both, then we can avoid these extra checks

According to the spec, there is a session variable to control the control mode or version. here I do not implement it first. 

@glorv  BTW, will parser to check it need to change the syntax? could you give a example?

Reply:
https://github.com/glorv/tidb/tree/rg-parser How about change the resource group syntax like this so it will only accept   either RU or raw definition but not both.

Reply:
<img width="650" alt="image" src="https://user-images.githubusercontent.com/5196885/210510377-b4122452-b6e3-458d-9238-c8ef3050300d.png">

Here is a snapshot of the parse result with this change.

Reply:
@glorv  I add a check before doing actually ddl job, the errors may clearer than yours? ptal

Reply:
ok. LGTM then.